src/store/slices/ui.ts

```
@@ -1058,9 +1060,22 @@ export const createUISlice: StateCreator<CairnStore, [], [], UISlice> = (
   // ── Agent config ───────────────────────────────
   setAgentConfig(patch) {
     set((s) => {
-      const next = { ...s.agentConfig, ...patch };
+      let next = { ...s.agentConfig, ...patch } as typeof s.agentConfig & { mode?: import("../../../shared/agent/approval-mode").Mode };
```

ESLint

[error] 'next' is never reassigned. Use 'const' instead.

(prefer-const)